api_v3/services/SessionService.php

```diff
@@ -90,8 +90,9 @@ function impersonateAction($secret, $impersonatedPartnerId, $userId = "", $type
 	{
 		KalturaResponseCacher::disableCache();
 		
+		$validatedExpiry = $expiry;
 		// verify that partnerId exists and is in correspondence with given secret
-		$result = myPartnerUtils::isValidSecret($partnerId, $secret, "", $expiry, $type);
+		$result = myPartnerUtils::isValidSecret($partnerId, $secret, "", $validatedExpiry, $type);
 		if ($result !== true)
```

Comment:
i think we also need to add check like we do on
alpha/apps/kaltura/lib/webservices/kSessionUtils.class.php:77